src/python/pants/init/logging.py

```diff
@@ -182,6 +183,11 @@ def initialize_stdio(global_bootstrap_options: OptionValueContainer) -> Iterator
         sys.__stdin__, sys.__stdout__, sys.__stderr__ = sys.stdin, sys.stdout, sys.stderr
 
 
+def pants_log_path(workdir: Path) -> Path:
```

Comment:
Can this be PurePath? Path is only if you need to make FS side effects like reading a file or deleting it.